### PR TITLE
[WIP] App Icon

### LINKED ISSUE #867: App Icon component
# Component name
AppIcon

> The Thumbnail component is primarily used to display product thumbnails, but this component has been leveraged to display other images, like app icons in places. 

## Components dependencies
This component could extend the existing thumbnail component, the difference being that it has no box-shadow (which looks like a light grey border)

"The pattern for apps in Shopify doesn’t have a border around the icon, following the styling of the app section and embedded apps." 
- https://shopify.slack.com/archives/C4Y8N30KD/p1547227607125700

### PR DESCRIPTION
**Pull request by laranoell**
### WHY are these changes introduced?

Resolves #867

### WHAT is this pull request doing?
-Creates a new `<AppIcon>` component 

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page} from '@shopify/polaris';

interface State {}

export default class Playground extends React.Component<never, State> {
  render() {
    return (
      <Page title="Playground">
        {/* Add the code you want to test here */}
      </Page>
    );
  }
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)

### CODE REVIEW

File: src/components/AppIcon/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+/>
+```
+
+### Large app icon
+
+Use when a app icon is a major focal point. Avoid this size in lists of like items.
+
+```jsx
+<AppIcon
+  source="https://cdn.shopify.com/s/files/applications/21d07b9a03ab6e538a053381def7b15d_80x80.png?1533144819"
```
Comment (kaelig) [inline src/components/AppIcon/README.md]: Should we encourage specifying the `_160x160.png` version of the icon, for hidpi support?

Comment (dpersing) [reply src/components/AppIcon/README.md]: Hi @laranoell! Any thoughts on the icon size?

Comment (laranoell) [reply src/components/AppIcon/README.md]: cc: @kaelig @dpersing

**Comment by dpersing** [comment]
@kaelig Can you take another look?

**Comment by kaelig** [comment]
@dpersing looks like the change to support hidpi is still pending. Happy to review again when it's applied.

**Comment by yourpalsonja** [comment]
Hey @laranoell ,
This repo getting a makeover! We want to make it easier to find existing issues, remove duplicates, and increase community engagement. If you issue is closed, re-labelled, or moved, don’t panic–it may be a duplicate (check the comments), or there is a new label or repository that work better.

If you disagree with the changes or have any other thoughts on how we can make Polaris better, please let us know!

I'm removing `WIP` and adding a `Development` label instead. We've got this issue in the `WIP` column in the Polaris backlog! What can we do to help this issue get merged? 

CC @kaelig @dpersing

**Comment by laranoell** [comment]
will revisit in the future

**Pull request closed without merging**